app/Authenticatable/HasParentModel.php

```diff
@@ -0,0 +1,41 @@
+<?php
```

Comment:
Is this trait necessary? doesn't a simple extend work?
Maybe it is adding to much complexity to just avoiding the two scopes (by now)

Reply:
Without the trait the relationships would look for `admin_id` and `assistant_id` unless we specify `user_id` in the User model.

I understand it seem a little complex but it's something that doesn't require maintenance and makes the code overall simple as we start to use the proper models more often. Also easier to read/understand.

Feel free to close the PR if you think this is not something you want right now.

You can read more about this approach here:https://tighten.co/blog/extending-models-in-eloquent

Reply:
Great explanation on that blog, 
I'm sure it can help us in the future to move the "administrative" and "assistive" tasks to its models

Thanks!